mobile_verifier/src/reward_shares.rs

```diff
@@ -100,20 +95,18 @@ impl TransferRewards {
         //
         // scale = [ 0.4 * total_emissions_pool ] / data_transfer_reward_sum
         //
-        let scale = if data_transfer_reward_sum / total_emissions_pool
-            > MAX_DATA_TRANSFER_REWARDS_PERCENT
+        let reward_scale = if reward_sum / total_emissions_pool > MAX_DATA_TRANSFER_REWARDS_PERCENT
         {
-            MAX_DATA_TRANSFER_REWARDS_PERCENT * total_emissions_pool / data_transfer_reward_sum
+            MAX_DATA_TRANSFER_REWARDS_PERCENT * total_emissions_pool / reward_sum
         } else {
             Decimal::ONE
         };
-        let remaining_poc_rewards = rewards_for_poc_and_dc - scale * data_transfer_reward_sum;
 
-        Ok(Self {
-            scale,
+        Self {
+            reward_scale,
             rewards,
-            remaining_poc_rewards,
-        })
+            reward_sum: reward_sum * reward_scale,
```

Comment:
in the event data transfer rewards _did_ exceed the allowed 40% at this point haven't you already allocated the overage to each gateway pubkey in the hashmap that made up the `reward_sum` total so it's too late to multiply by the scale percentage here? like, if we tallied up all of the individual data transfer rewards at the end and found they exceeded the allowable amount shouldn't we iterate over each entry in the map, multiplying _that_ by the reward_scale percentage and then re-sum the total?